### ISSUE TITLE
Infinite loop in TextEditor.getLongTitle() causes editor to hang

### ISSUE THREAD

**Issue by davewasmer** [atom/atom#10060]
`getLongTitle()` appears to be responsible for disambiguating the tab's title in the case of multiple tabs with the same file name. The code will hit an infinite loop when presented with a certain directory structure:

```
alpha/
  foo/
    bar.txt
beta/
  foo/
    bar.txt
  something-else/
    bar.txt
```

If all three `bar.txt` files are opened in Atom, when the third one is opened and `getLongTitle()` is called, it will hit an infinite loop.

The method takes the file path for the tab being opened, and compares it to the others already open. If it finds colliding base filenames (i.e. `bar.txt`), and attempts to find the first parent directory that differs by stripping off one segment from each file path, until it finds a unique directory. The problem is that by comparing the paths this way, one level at a time, you can have one path collide at one level, and another path collide at a different level, resulting in no unique directory for all paths being found.

The problem becomes apparent when we step through the logic while attempting to open `beta/foo/bar.txt` with the other two `bar.txt` files already open.

1. All three file paths are selected for disambiguation since they all have the same basename (`bar.txt`).
2. Strip one layer off (now at `beta/foo`), and check if the lowest directory is unique. It's not, because of `alpha/foo/bar.txt` (whose path is now `alpha/foo`, colliding with `beta/foo`).
3. Strip off another layer (now at `beta`), and check if the lowest directory is unique. It's not, because of `beta/something-else/bar.txt` (whose path is now `beta`).
4. Repeat, but now all the paths are the same because we hit the root of the project.

Eventually, all the paths are reduced to `/`, and the while loop gets stuck.

Happy to try submitting a PR, but feels like this could use some discussion about what the actual goal is. I'm not sure the current algorithm would work, even with a different implementation. There is no single ancestor directory that could disambiguate all three file paths, so it seems like a different approach is needed.

Here's a zip with a reproduction project:
[atom-bug-repro.zip](https://github.com/atom/atom/files/61770/atom-bug-repro.zip)

**Comment by 50Wliu** [comment]
I believe this has already been fixed in Atom 1.4.0-beta0 - it would be great if you could try that and report back.  Thanks!

**Comment by davewasmer** [comment]
Just checked master and saw that - downloading the beta now :wink:

**Comment by davewasmer** [comment]
Looks good :+1:

**Comment by Peekmo** [comment]
Hello,

I have that problem too, and I would like to know if there's a fix for 1.3 users please ? 'Cause my package does not work anymore and add crashes for some users..

Thank you :+1:
